### PR TITLE
[ll] Update bitflags to 1.0

### LINKED ISSUE #1557: Update bitflags to 1.0

### PR DESCRIPTION
**Pull request by davll**
Closes #1557 

## TODOs:

- [x] Add constants back to avoid breaking existing codes

### CODE REVIEW

**Comment by kvark** [comment]
This would be nice to do for `draw_state` first, assuming we want to use it. Or better - first figure this out :)
@msiglreith could you express your earlier proposals for `draw_state` in an issue to discuss?

File: src/hal/src/buffer.rs [hunk 1]
```diff
@@ -48,66 +48,110 @@ impl Error for ViewError {
...
-        const INDIRECT = 0x80,
+        const INDIRECT = 0x80;
         ///
-        const VERTEX = 0x100,
+        const VERTEX = 0x100;
     }
 );
 
+///
+pub const TRANSFER_SRC: Usage   = Usage::TRANSFER_SRC;
```
Comment (kvark) [inline src/hal/src/buffer.rs]: this is great! finally, we have namespaces for bit flags :)

Comment (davll) [reply src/hal/src/buffer.rs]: Yes, this makes code cleaner. Maybe we can rename some flags properly later.

**Comment by davll** [comment]
@kvark
Yes, reexported constants are temporary for backward compatibility. We can remove them later.

**Comment by kvark** [comment]
Let's make CI happy and proceed then?

**Comment by davll** [comment]
Yeah~

**Comment by davll** [comment]
OK, I think the PR is ready to merge

**Comment by msiglreith** [comment]
> @msiglreith could you express your earlier proposals for draw_state in an issue to discuss?

Everything contained in here https://github.com/gfx-rs/gfx/issues/1509#issuecomment-330813605. We can discuss it further tho.

**Comment by davll** [comment]
@kvark Done

**Pull request merged**